webui/react/src/components/InteractiveTable.tsx

```diff
@@ -70,8 +66,7 @@ interface ContextMenuProps {
 export interface ColumnDef<RecordType> extends ColumnType<RecordType> {
   dataIndex: string;
   defaultWidth: number;
-  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-  isFiltered?: (s: any) => boolean; // any extends Settings
+  isFiltered?: (s: unknown) => boolean;
```

Comment:
suggestion: I think this can be something like `isFiltered?: <T extends InteractiveTableSettings>(s: T) => boolean`

Reply:
tried a few variations of this, no luck.

it's definitely the right path though. I found out it's possible to use generic types in such a way that you can use the type parameters in the function body, e.g.

```
const InteractiveTable = <RecordType, Settings extends InteractiveTableSettings>({
  loading,
  scroll,
  dataSource,
  columns,
  containerRef,
  settings,
  updateSettings,
  ContextMenu,
  areRowsSelected,
  ...props
}: InteractiveTableProps<RecordType, Settings>): JSX.Element => {
```

then the specifics for `Settings` should in theory get inferred by virtue of:
```
interface InteractiveTableProps<RecordType, Settings> extends TableProps<RecordType> {
  ContextMenu?: React.FC<ContextMenuProps>;
  areRowsSelected?: boolean;
  columns: ColumnDef<RecordType>[],
  containerRef: MutableRefObject<HTMLElement | null>,
  settings: Settings;
  updateSettings: UpdateSettings<Settings>
}
```

and you can do things like 

`const column: ColumnDef<RecordType> = ...`

even though in this case `RecordType` is just a type parameter of the function.

I think ultimately this is the way to go, I tried it a bit [here](https://github.com/determined-ai/determined/pull/4291) and was able to git rid of all the `any` and type errors within `InteractiveTable` but it breaks downstream consumption, and I'm tired of messing with it for now. 

So just gonna merge and come back later.